openmp/libomptarget/plugins-nextgen/amdgpu/src/rtl.cpp

```diff
@@ -3109,6 +3126,47 @@ struct AMDGPUPluginTy final : public GenericPluginTy {
     return HSA_STATUS_ERROR;
   }
 
+  /// Detect if current architecture is an APU.
+  void isAPU() {
+    char GfxName[64];
+
+    if (!KernelAgents.size())
+      return;
+
+    // Do not allow for mixed APU+discrete GPU combinations: an APU can only be
+    // that, so only check the first GPU agent.
```

Comment:
I don't understand why we assume you can't have an MI250X and a MI300A in the same system.

Reply:
Replied above, will fix.